libse/NetflixQualityCheck/NetflixCheckMaxCps.cs

```diff
@@ -9,20 +10,20 @@ public class NetflixCheckMaxCps : INetflixQualityChecker
         /// </summary>
         public void Check(Subtitle subtitle, NetflixQualityController controller)
         {
+            // constant
+            string comment = "Maximum " + controller.CharactersPerSecond + " characters per second";
             var oldIgnoreWhiteSpace = Configuration.Settings.General.CharactersPerSecondsIgnoreWhiteSpace;
             try
             {
                 foreach (Paragraph p in subtitle.Paragraphs)
                 {
-                    var charactersPerSeconds = Utilities.GetCharactersPerSecond(p);
+                    int textLenExludingNonDisplayChars = GetTextLengthExcludingNonDisplayChars(p.Text);
+                    double charactersPerSeconds = textLenExludingNonDisplayChars / p.Duration.TotalSeconds;
                     if (charactersPerSeconds > controller.CharactersPerSecond)
                     {
                         var fixedParagraph = new Paragraph(p, false);
-                        while (Utilities.GetCharactersPerSecond(fixedParagraph) > controller.CharactersPerSecond)
-                        {
-                            fixedParagraph.EndTime.TotalMilliseconds++;
-                        }
```

Comment:
not really fan of this code 😕 ... the newer will do all the work on a constant time